### PR TITLE
feat(google): add support for Google SQL

### PR DESCRIPTION
**Pull request by ekifel**
Issue #386

Output example: 

```
  NAME                                          MONTHLY QTY  UNIT       PRICE   HOURLY COST  MONTHLY COST
  google_sql_database_instance.my_instance                                                                 
  ├─ vCPUs (regional)                                11,680  hours      0.0826       1.3216      964.7680  
  ├─ Memory (regional)                               43,800  GB         0.0140       0.8400      613.2000  
  ├─ Storage (SSD, regional)                            100  GB-months  0.3400       0.0466       34.0000  
  ├─ Backups                                          1,000  GB-months  0.0800       0.1096       80.0000  
  ├─ IP address (if unused)                             730  hours      0.0100       0.0100        7.3000  
  └─ Replica                                    
     ├─ vCPUs (zonal)                                11,680  hours      0.0413       0.6608      482.3840  
     ├─ Memory (zonal)                               43,800  GB         0.0070       0.4200      306.6000  
     └─ Storage (SSD, zonal)                            100  GB-months  0.1700       0.0233       17.0000  
  Total                                                                              3.4319    2,505.2520
```

```
  NAME                                          MONTHLY QTY  UNIT       PRICE   HOURLY COST  MONTHLY COST
  google_sql_database_instance.my_instance                                                                 
  ├─ SQL instance (db-n1-standard-32, zonal)            730  hours      2.1620       2.1620    1,578.2600  
  ├─ Storage (SSD, zonal)                                10  GB-months  0.1700       0.0023        1.7000  
  └─ Backups                                          1,000  GB-months  0.0800       0.1096       80.0000  
  Total                                                                              2.2739    1,659.9600 
```

### CODE REVIEW

**Review by alikhajeh1 — CHANGES_REQUESTED** [review]
@sinabakh thanks for starting this and @ekifel many thanks for pushing it over the finish line. This was not easy to review as the pricing is damn complicated :D here are my suggestions:
1. Add this for MySQL/Postgres/SQL Server: a cost component name `IP address (if unused)`, units `hours`, using the `AttributeFilters.description` filter `IP address idling in seconds` ($0.01/hr). This should only be shown if `settings.ip_configuration` is defined and it has `ipv4_enabled`. Don't show this for the replica sub-resource (see item 6 below).

2. Refactor the "CPU Credits", Memory and SQL Instance cost components to follow these rules:

  - a) If the instance is db-f1-micro/db-g1-small, show cost component name like `SQL instance (db-f1-micro, zonal)`. Units should be "hours".

  - b) If the instance is high availability db-f1-micro/db-g1-small, show cost component name like `SQL instance (db-f1-micro, regional)`. Units should be "hours".

  - c) If instance if custom type, show cost component names "vCPUs (zonal)", with units "hours" and "Memory (zonal)" (units of GB is fine).

  - d) If instance if high availability custom type, show cost component names "vCPUs (regional)", with units "hours" and "Memory (regional)" (units of GB is fine).

3. Storage should show "Storage (zonal)" or "Storage (regional)".

4. Microsoft SQL server does not support db-f1-micro/db-g1-small, so it should have a cost component name "License (web)", "License (standard)", or "License (enterprise)" with units "hours". Express license is free so no License cost should be shown if that's used.

    This should also be followed (minimum of 4 vCPUs) in the calculation for this cost component (from the GCP pricing page):
  "Instances with fewer than 4 vCPUs will be charged for SQL Server at 4 times the license rate to comply with these requirements. For instances with 4 or more vCPUs, you will be charged for the number of SQL Server licenses that is equal to the number of vCPUs."

    The license cost component does not double for REGIONAL, as the pricing page mentions "High availability, or regional instances, will only incur the cost for a single license for the active resource". However, it should be included for the replica if one is defined (see item 6).

5. Test the logic works (using unit tests) for other regions, e.g. I get 0s for this (maybe I'm wrong?):
```
provider "google" {
 credentials = "{\"type\":\"service_account\"}"
 region      = "us-west-4"
}
```

6. Add the ZONAL versions of vCPU, Memory, SQL instance, Storage and License cost components for replica sub-resource. From GCP pricing page "Read replicas and failover replicas are charged at the same rate as stand-alone instances". See https://github.com/infracost/infracost/blob/master/internal/providers/terraform/aws/instance.go#L39-L40 for an example of how to add sub-resource pricing.
```
resource "google_sql_database_instance" "custom_postgres" {
  name             = "master-instance"
  database_version = "POSTGRES_11"
  settings {
    tier = "db-custom-16-61440"
  }
  replica_configuration {
    username = "replica"
  }
}
```

7. Add the following to the usage-file and use it to decide if 1 yr or 3 yr commitment price should be used for the vCPUs and Memory (defaults to none if it's not defined, meaning on-demand price applies).
```
commitment_term: none # Purchase commitment term, can be: none, 1_year, 3_year.
```

8. Add unit tests to cover the above use-cases :)

**Comment by sinabakh** [comment]
@ekifel Thanks for finishing it, it was so complicated to connect all the cost components and logic. 🚀

**Comment by ekifel** [comment]
@alikhajeh1 this is ready to review!

**Comment by alikhajeh1** [comment]
Thanks! This is almost done... a few small issues left:

Regarding #2 above, the `SQL instance` cost component should only show for db-f1-micro/db-g1-small. For all other instance types, only the vCPUs and Memory cost component should be shown. At the moment, I see all 3 are shown for this:
```
resource "google_sql_database_instance" "custom_postgres" {
  name             = "master-instance"
  database_version = "POSTGRES_11"

  settings {
    tier = "db-custom-1-3750"
  }
}
```

Regarding `Backups`, this is a usage-based price, so it should appear with a price (and no costs) if I do this:
```
resource "google_sql_database_instance" "small_mysql" {
		name             = "master-instance"
		database_version = "MYSQL_8_0"	
		settings {
			tier = "db-g1-small"
		}
}
```

Can you please add unit tests to cover the above 2 issues, as well as other params that have no tests, e.g. I don't see a test case looking for `PD_HDD`.

File: internal/providers/terraform/google/sql_database_instance.go [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+	PostgreSQL
+	SQLServer
+)
+
+func GetSQLInstanceRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "google_sql_database_instance",
+		RFunc: NewSQLInstance,
+		Notes: []string{
+			"Cloud SQL network/SQL Server license/1-3 years commitments costs are not yet supported.",
```
Comment (alikhajeh1) [inline internal/providers/terraform/google/sql_database_instance.go:25]:
```suggestion
			"Cloud SQL network, SQL Server license, 1-3 years commitments costs are not yet supported.",
```

**Comment by ekifel** [comment]
@alikhajeh1, this is ready to review. Added an example for MySQL (db-n1-...) to PR description, + checked it in Google Calculator, costs are equals

**Pull request merged**